src/components/ReservationPage.tsx

```diff
@@ -0,0 +1,395 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useContext,
+  useState,
+} from "react";
+import TextField from "@material-ui/core/TextField";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormControl from "@material-ui/core/FormControl";
+import FormLabel from "@material-ui/core/FormLabel";
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import {
+  IconButton,
+  makeStyles,
+  Dialog,
+  Toolbar,
+  Typography,
+  DialogActions,
+  DialogContent,
+  Button,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import Select from "./Select";
+import { AuthContext } from "./AuthContext";
+import { makeTransition } from "./Transition";
+import UserGroup from "../user/UserGroup";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+
+const useStyles = makeStyles(() => ({
+  root: {
+    display: "flex",
+    alignItems: "center",
+  },
+  right: {
+    display: "flex",
+    justifyContent: "flex-end",
+  },
+  center: {
+    display: "flex",
+    justifyContent: "center",
+  },
+  guests: {
+    display: "none",
+  },
+  notes: {
+    display: "none",
+  },
+  left: {
+    display: "flex",
+    justifyContent: "flex-start",
+    margin: 0,
+  },
+  list: {
+    display: "flex",
+    justifyContent: "flex-start",
+    alignContent: "row",
+  },
+  test: {
+    margin: 0,
+  },
+  container: {
+    flexWrap: "wrap",
+  },
+  textField: {
+    width: 200,
+  },
+  alignSelect: {
+    marginTop: "auto",
+  },
+  text: {
+    paddingTop: "10px",
+  },
+}));
+
+const transition = makeTransition("left");
+const initialGroups: UserGroup[] = [];
+
+const ReservationPage: FunctionComponent<CalendarUIProps> = ({
+  dispatch,
+  state,
+}) => {
+  const [isSubmitionCompleted, setSubmitionCompleted] = useState(false);
+  const { user } = useContext(AuthContext);
+  const [groups, setGroups] = useState(initialGroups);
+  const projects = state.projects.filter((project) =>
+    user?.projectIds.includes(project.id)
+  );
+  const initialProject = projects[0];
+  const [currentProject, setCurrentProject] = useState(initialProject);
+
+  useEffect(() => {
+    fetch(`/api/project_groups/${currentProject?.id}`)
+      .then((response) => response.json())
+      .then((groups) => {
+        const usergroups = groups.map(
+          (group: UserGroup) => new UserGroup(group)
+        );
+        setGroups(usergroups);
+      })
+      .catch(console.error);
+  }, [currentProject, user]);
```

Comment:
Interesting bit to think about this one.  I don't think we want to hit this API upon opening the reservation form.  This should _probably_ be in state somewhere already.  I'll chart it out and come back with a better answer.

Reply:
that would probably require that the `currentProject` in ReservationPage.tsx be linked to the calendar state, right? since we would need to get the useEffect to update as it is currently at line 104 here?

Reply:
### Update on fetching in the reservation form
Upon review, I want to
- move `state.currentProjectGroups` to `state.groups` (user receives all groups they are authorized to see; may need to amend this idea, but this will work for now)
- move the fetching of groups from within the project dashboard to the parent `<Calendar>` component.

`<ProjectDashboard>` and `<ReservationForm>` will just filter and consume `state.groups` as needed.  `<ProjectDashboard>` still determines what the "current" project & group are.  `<ReservationForm>` will read the current project & group, but will fallback to some reasonable default if those are unset.